students/perrymant/blog-01-hello-world-deconstructed.md

```diff
@@ -0,0 +1,347 @@
+```java
++---Frog.java---------------+
+|                           |
+|  public class Frog {      |
+|  }                        |
+|                           |
+|  class FrogFeed { |
+|  }                        |
++---------------------------+
+```
+#### 2. `protected` access modifier
+
+The `protected` access modifier can't actually be applied to classes, but it can be used with methods, constructors and data members, so we'll look at it here, and start by understanding how we organise classes in packages and files.
+
+The way we organise our classes is through files, packages and folders.
+We've already seen how .java files contain the classes, and you're probably familiar with folders (or directories) because your computer's Operating System uses these to create a hierarchy for storing files.
+Packages are exactly the same as folders, with the only difference being that they are the folders within the `src` folder:
+```java
+Documents
+└── pizzeria                                <- folder
+    └── src                                 <- folder
+        └── com                             <- package
+            └── example                     <- package
+                ├── appliances              <- package
+                │   ├── Oven.java
+                │   └── Refrigerator.java
+                ├── food                    <- package
+                │   ├── Antipasti.java
+                │   ├── Pasta.java
+                │   ├── Pizza.java
+                │   └── ingredients         <- package
+                │       ├── Cheese.java
+                │       ├── Flour.java
+                │       └── Tomato.java
+                └── restaurant              <- package
+                    ├── Kitchen.java
+                    └── Toilet.java
+```
+Packages are used for organising related classes, and to put a class in a package, you need to add the line
+
+`package `*packagename*`;`
+
+as the very first instruction in the source file.
+Typically you will see these in a format that has one or more identifiers separated by periods such as `java.util.random`.
+The naming format here is trying to make sure that each name is unique, so as to avoid namespace collisions and is the same as with import naming.
+For example, if I wanted to make use of the class `List`, I would find that there are quite a few:
+![Namespace](assets/blog-01-hello-world-deconstructed/namespace.png)
+*which one do I want?* I have to specify, and this reversed naming format helps keep all of these classes and packages separate.
+
+When the `protected` access modifier is used, it signals that the method, constructors or data member is accessible both within and outside the package but through inheritance only.
+
+#### 3. `package private` access modifier
+
+When no named package statements are included in a source file, then the classes will be placed in the *default package*.
```

Comment:
I think we might be conflating two unrelated terms here.

The default package, i.e. the package your code lives in if you don't create a package and don't have the `package com.whatever` at the top, is entirely unrelated to the default (package private) access modifier. This default package is generally discouraged. On the other hand, I encourage the use of the default access modifier.